multiversx_sdk/abi/serializer_test.py

```diff
@@ -143,6 +144,27 @@ def test_serialize():
 
     assert data == "41@42@43"
 
+    # counted-variadic, of different types
```

Comment:
Indeed, we don't have this restriction. However, in practice, people should not use this "feature".